src/format.ts

```diff
@@ -110,8 +122,127 @@ function checkFormat(srcFiles: string[], baseArgs: string[]): Promise<boolean> {
     out.on('data', (data: Buffer) => {
       output += data;
     });
-    out.on('end', () => {
+
+    out.on('end', async () => {
+      const files = output.split('<?xml version=\'1.0\'?>\n');
+      for (let i = 1; i < files.length; i++) {
+        const replacements = getReplacements(files[i]);
+        if (replacements.length > 0) {
+          const diff = await getDiffObj(srcFiles[i - 1], replacements)
+          printDiffs(diff, options);
+        }
+      }
       resolve(output.indexOf('<replacement ') === -1 ? true : false);
     });
   });
 }
+
+/**
+ * Parses through xml string for the replacement string, offset, length.
+ *
+ * @param output xml string from clangFormat
+ */
+function getReplacements(output: string): replacement[] {
+  const replacements: replacement[] = [];
+  const offsets: string[]|null = output.match(/(?<=offset=\')(\d+)(?=\')/g);
+  const lengths: string[]|null = output.match(/(?<=length=\')(\d+)(?=\')/g);
+  const fixes: string[]|null =
+      output.match(/(?<=length=\'\d+\'>)(.*)(?=<\/replacement>)/g);
+  if (lengths === null || offsets === null || fixes === null) {
+    return replacements;
+  }
+  for (let i = 0; i < offsets.length; i++) {
+    replacements[i] = {
+      offset: Number(offsets[i]),
+      length: Number(lengths[i]),
+      fix: entities.decodeXML(fixes[i])
+    };
+  }
+  return replacements;
+}
+
+/**
+ * Gets an object containing the differences between the original file and after
+ * changes have been made
+ *
+ * @param file
+ * @param replacements array of all the formatting issues within in the file
+ */
+async function getDiffObj(
+    file: string, replacements: replacement[]): Promise<jsdiff.IUniDiff> {
+  const text = await readFilep(file, 'utf8');
+
+  const fixed = performFixes(text, replacements);
+  const diff = jsdiff.structuredPatch(
+      file, 'no new file', text, fixed, 'no old header', 'no new header',
+      {context: 3});
+  jsdiff.applyPatch('diff', diff);
+  return Promise.resolve(diff);
```

Comment:
Change this to simply `return diff`. Async functions automatically wrap return values in a promise.

Reply:
done